src/uct/ib/rc/base/rc_iface.c

```diff
@@ -150,39 +150,52 @@ ucs_status_t uct_rc_iface_query(uct_rc_iface_t *iface,
                                   UCT_IFACE_FLAG_EVENT_SEND_COMP |
                                   UCT_IFACE_FLAG_EVENT_RECV;
 
-    if (dev->atomic_arg_sizes & sizeof(uint64_t)) {
-        /* TODO: remove deprecated flags */
-        iface_attr->cap.flags              |= UCT_IFACE_FLAG_ATOMIC_DEVICE;
+    if (dev->pci_fadd_arg_sizes || dev->pci_cswap_arg_sizes) {
+        if (dev->pci_fadd_arg_sizes & sizeof(uint64_t)) {
+            iface_attr->cap.atomic64.op_flags  |= UCS_BIT(UCT_ATOMIC_OP_ADD);
+            iface_attr->cap.atomic64.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_ADD);
+        }
+        if (dev->pci_cswap_arg_sizes & sizeof(uint64_t)) {
+            iface_attr->cap.atomic64.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_CSWAP);
+        }
+#if HAVE_IB_EXT_ATOMICS
+        if (dev->pci_fadd_arg_sizes & sizeof(uint32_t)) {
+            iface_attr->cap.atomic32.op_flags  |= UCS_BIT(UCT_ATOMIC_OP_ADD);
+            iface_attr->cap.atomic32.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_ADD);
+        }
+        if (dev->pci_cswap_arg_sizes & sizeof(uint32_t)) {
+            iface_attr->cap.atomic32.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_CSWAP);
+        }
+#endif
+        iface_attr->cap.flags                  |= UCT_IFACE_FLAG_ATOMIC_CPU;
```

Comment:
push this inside each of the 'if' -s above

Reply:
We still use (md->dev.pci_fadd_arg_sizes || md->dev.pci_cswap_arg_sizes) as a check for pci atomic in uct_rc_mlx5_iface_fence(). Maybe return UCT_IB_DEVICE_FLAG_PCI_ATOMICS?

Reply:
where is this flag defined?

Reply:
Was added here https://github.com/openucx/ucx/pull/3226/commits/f5c3cc5cb734c419574c642d777e6847ad1b0751, removed here https://github.com/openucx/ucx/pull/3226/commits/42be17a8d93f996d67dce32187189eb5e13b84e3

Reply:
this comment is not addressed

Reply:
since if (<pci atomics>) is true iff we work in pci atomics mode no need to push inside each "if" now